engine/api/get-started.md

```diff
@@ -606,6 +608,109 @@ $ curl --unix-socket /var/run/docker.sock \
 </div>
 </div> <!-- end tab-content -->
 
+
+### Pull images with authentication
+
+Pull images, like `docker pull`, with authentication:
+
+> **Note**: Credentials are sent in the clear. Docker's official registries use
+> HTTPS. Private registries should also be configured to use HTTPS.
+
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" data-target="#tab-pullimages-auth-python" data-group="python">Python</a></li>
+  <li><a data-toggle="tab" data-target="#tab-pullimages-auth-go" data-group="go">Go</a></li>
+  <li><a data-toggle="tab" data-target="#tab-pullimages-auth-curl" data-group="curl">Curl</a></li>
+</ul>
+
+<div class="tab-content">
+<div id="tab-pullimages-auth-python" class="tab-pane fade in active" markdown="1">
+
+The Python SDK retrieves authentication information from the [credentials
+store](/engine/reference/commandline/login/#credentials-store) file and
+integrates with [credential
+helpers](https://github.com/docker/docker-credential-helpers){: target="_blank"
+class="_" }. It is possible to override these credentials, but that is out of
+scope for this Getting Started guide. After using `docker login`, the Python SDK
+uses these credentials automatically.
+
+```python
+import docker
+client = docker.from_env()
+image = client.images.pull("alpine")
+print image.id
+```
+
+</div>
+
+<div id="tab-pullimages-auth-go" class="tab-pane fade" markdown="1">
+
+```go
+package main
+
+import (
+  "io"
+  "os"
+  "encoding/json"
+  "encoding/base64"
+
+  "github.com/docker/docker/api/types"
+  "github.com/docker/docker/client"
+  "golang.org/x/net/context"
+)
+
+
+func main() {
+  ctx := context.Background()
+  cli, err := client.NewEnvClient()
+  if err != nil {
+    panic(err)
+  }
+
+  authConfig := types.AuthConfig{
+      Username: "username",
+      Password: "password",
+  }
+  encodedJSON, err := json.Marshal(authConfig)
+  if err != nil {
+      return err
```

Comment:
The `main` function does not return an error. This should be `panic(err)` instead